### PR TITLE
ci: adds release automation

### LINKED ISSUE #27: Add PyPI Release CI
See https://github.com/marketplace/actions/pypi-publish for publishing with the OIDP mechanism.

### PR DESCRIPTION
**Pull request by tdstein**
Resolves #27

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+An beta pre-release. Created to support closed external user testing. `N` starts at **1** and increments by 1 (`X.Y.beta1`, `X.Y.beta2`, ..., `X.Y.betaN`).
+
+*https://peps.python.org/pep-0440/#pre-releases*
+
+
+**`X.Y.rcN`**
+
+An release-candidate pre-release. Created to support open external user testing. `N` starts at **1** and increments by 1 (`X.Y.rc1`, `X.Y.rc2`, ..., `X.Y.rcN`).
+
+*https://peps.python.org/pep-0440/#pre-releases*
```
Comment (mmarchetti) [inline CONTRIBUTING.md:108]: The [Python version spec](https://packaging.python.org/en/latest/specifications/version-specifiers/#public-version-identifiers) doesn't adhere to semver syntax for pre-releases (though it does for dev releases). For example, `X.YbN` is the syntax for a beta release.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  default:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v5
+    - run: make
+    - id: release
+      uses: pypa/gh-action-pypi-publish@release/v1
```
Comment (mmarchetti) [inline .github/workflows/release.yaml:16]: No secrets - nice!

File: CONTRIBUTING.md (hunk 1 above)
Comment (tdstein) [reply CONTRIBUTING.md:108]: Fixed

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+A stable minor release. Created for added functionality in a backward compatible manner. `N` starts at **0** and increments by 1 (`X.0.Z`, `X.1.Z`, ..., `X.N.Z`).
+
+*https://semver.org*
+
+**`N.Y.Z`**
+
+A stable major release. Created for incompatbile API changes. `N` starts at **0** and increments by 1 (`0.Y.Z`, `1.Y.Z`, ..., `N.Y.Z`).
+
+*https://semver.org*
```
Comment (nealrichardson) [inline CONTRIBUTING.md:126]: I don't think we have to explain semantic versioning here.

```suggestion
```

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name: Release
+on:
+  push:
+    tags:
+      - "v*.*.*"
```
[the remark this answers is not in the document]
Comment (tdstein) [reply .github/workflows/release.yaml:5]: pre-release tags are marked in PyPI https://pypi.org/project/posit-sdk/#history

**Review by nealrichardson — APPROVED** [review]

**Pull request merged**